Source/WebCore/style/ScopedName.h

```diff
@@ -47,3 +48,30 @@ inline WTF::TextStream& operator<<(WTF::TextStream& ts, ScopedName scopedName)
 
 } // namespace Style
 } // namespace WebCore
+
+namespace WTF {
+
+using WebCore::Style::ScopedName;
+using WebCore::Style::ScopeOrdinal;
+
+struct ScopedNameHash {
+    // FIXME: incorporate scope ordinal into the hash
+    static unsigned hash(const ScopedName& key) { return computeHash(key.name); }
+
+    // FIXME: compare the scope ordinal too.
+    static bool equal(const ScopedName& a, const ScopedName& b) { return a.name == b.name; }
```

Comment:
why not do these things already in this patch?

Reply:
We had a discussion about mechanically replacing things with `ScopedName` and you mentioned using the name hash as the hash, but yeah we could just hash the whole thing.

Reply:
It is probably nicer to either not have traits at all (and just keep hashing on the AtomString manually) or have proper traits.

As mentioned, I'm not sure how useful making these hashable is going to be in long run. The ordinals need to be resolved and replaced with scope roots before they can actually be compared for equality/looked up from a global hash.

Reply:
Is there a reason why we don't put the scope root pointer into `ScopedName`? Instead of having to resolve to the scope root when it's needed?